tests/test_subparser.py

```diff
@@ -0,0 +1,263 @@
+from dataclasses import dataclass, field
+from typing import Union
+from sys import version_info
+
+from pytest import mark, raises
+
+from pydargs import parse
+
+
+@dataclass
+class Action1:
+    a: int
+    b: str = "abc"
+
+
+@dataclass
+class Action2:
+    c: int = 42
+    b: str = "abc"
+    e: str = field(default="def", metadata=dict(positional=True))
+
+
+@dataclass
+class Action3:
+    d: float
+    e: list[str] = field(default_factory=lambda: ["def"])
+
+
+@dataclass
+class Action4:
+    sub_action: Union[Action1, Action2]
+    string4: str = "four"
+
+
+class TestParseAction:
+    @dataclass
+    class Config:
+        action: Union[Action1, Action2]
+        var: int = 12
+        flag: bool = field(default=False, metadata=dict(as_flags=True))
+
+    @mark.parametrize(
+        "help_string",
+        [
+            "usage: prog [-h] [--var VAR] [--flag | --no-flag]",
+            "{Action1,action1,Action2,action2} ...",
+            "action:  {Action1,action1,Action2,action2}",
+        ],
+    )
+    def test_help(self, capsys, help_string: str):
+        with raises(SystemExit):
+            parse(self.Config, ["--help"], prog="prog")  # type: ignore
+        captured = capsys.readouterr()
+        assert help_string in captured.out.replace("\n", "")
+
+    @mark.skipif(version_info < (3, 10), reason="python3.9 prints s slightly different help")
+    @mark.parametrize(
+        "help_string",
+        ["prog Action1 [-h] --a A [--b B]", "options:  -h"],
+    )
+    def test_action_help(self, capsys, help_string: str):
+        with raises(SystemExit):
+            parse(self.Config, ["Action1", "--help"], prog="prog")  # type: ignore
+        captured = capsys.readouterr()
+        assert help_string in captured.out.replace("\n", "")
+        assert "prog Action1 [-h] --a A [--b B]" in captured.out.replace("\n", "")
```

Comment:
is this also here on purpose? I think this is also part of the help_string parametrize right?